contracts/implement/ConcentratedLiquidity.sol

```diff
@@ -348,40 +368,94 @@ abstract contract ConcentratedLiquidity is IConcentratedLiquidity {
     }
 
     function collectFee(uint256 nftTokenId) public virtual {
-        address owner = _msgSender();
-        require(owner == positionDexNft.ownerOf(nftTokenId), "!Owner");
+        UserLiquidity.Data memory liquidityData = concentratedLiquidity[
+            nftTokenId
+        ];
+        (
+            uint256 feeBaseAmount,
+            uint256 feeQuoteAmount,
+            uint256 newFeeGrowthBase,
+            uint256 newFeeGrowthQuote
+        ) = _collectFee(
+                liquidityData.pool,
+                liquidityData.feeGrowthBase,
+                liquidityData.feeGrowthQuote,
+                liquidityData.liquidity,
+                liquidityData.indexedPipRange
+            );
+
+        address user = _msgSender();
+        withdrawLiquidity(
+            liquidityData.pool,
+            user,
+            SpotHouseStorage.Asset.Base,
+            feeBaseAmount
+        );
+
+        withdrawLiquidity(
+            liquidityData.pool,
+            user,
+            SpotHouseStorage.Asset.Quote,
+            feeQuoteAmount
+        );
+        liquidityData.feeGrowthBase = newFeeGrowthBase;
```

Comment:
`liquidityData` is memory, so `liquidityData.feeGrowthBase = newFeeGrowthBase;` will not update storage

Same with `liquidityData.feeGrowthQuote = newFeeGrowthQuote;`